### ISSUE TITLE
AttributeError: 'WebSocket' object has no attribute 'page_id' on wp.redirect

### ISSUE THREAD

**Issue by matthoendorf** [justpy-org/justpy#140]
Issue seems to be related to 0.1.4 change "use wp.redirect in request handlers with no need for page_ready event"

In our example, when a user navigates to a page (i.e. http://localhost:5000/overview?project_id=37), it first checks if they are logged in. If they aren't, we save the original ```request.url```, redirect them to AD login, and send them back to the JustPy application. Once they're set to logged in, we redirect the page to the original ```request.url```. However, the full URL gives an error, so we have to remove the base domain (i.e. http://localhost:5000).

This works:
```
users[request.session_id]['redirect_page'] = str(request.url).replace('http://localhost:5000','')
...
wp.redirect = users[request.session_id]['redirect_page']
```

But this:
```
users[request.session_id]['redirect_page'] = str(request.url)
...
wp.redirect = users[request.session_id]['redirect_page']
```

Causes an error, but still redirects correctly:
```
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/ubuntu/.local/share/virtualenvs/analysis-hub-OumnfNmP/lib/python3.6/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 154, in run_asgi
    result = await self.app(self.scope, self.asgi_receive, self.asgi_send)
  File "/home/ubuntu/.local/share/virtualenvs/analysis-hub-OumnfNmP/lib/python3.6/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/home/ubuntu/.local/share/virtualenvs/analysis-hub-OumnfNmP/lib/python3.6/site-packages/starlette/applications.py", line 111, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/ubuntu/.local/share/virtualenvs/analysis-hub-OumnfNmP/lib/python3.6/site-packages/starlette/middleware/errors.py", line 146, in __call__
    await self.app(scope, receive, send)
  File "/home/ubuntu/.local/share/virtualenvs/analysis-hub-OumnfNmP/lib/python3.6/site-packages/starlette/middleware/httpsredirect.py", line 19, in __call__
    await self.app(scope, receive, send)
  File "/home/ubuntu/.local/share/virtualenvs/analysis-hub-OumnfNmP/lib/python3.6/site-packages/starlette/middleware/gzip.py", line 20, in __call__
    await self.app(scope, receive, send)
  File "/home/ubuntu/.local/share/virtualenvs/analysis-hub-OumnfNmP/lib/python3.6/site-packages/starlette/exceptions.py", line 58, in __call__
    await self.app(scope, receive, send)
  File "/home/ubuntu/.local/share/virtualenvs/analysis-hub-OumnfNmP/lib/python3.6/site-packages/starlette/routing.py", line 566, in __call__
    await route.handle(scope, receive, send)
  File "/home/ubuntu/.local/share/virtualenvs/analysis-hub-OumnfNmP/lib/python3.6/site-packages/starlette/routing.py", line 283, in handle
    await self.app(scope, receive, send)
  File "/home/ubuntu/.local/share/virtualenvs/analysis-hub-OumnfNmP/lib/python3.6/site-packages/starlette/endpoints.py", line 76, in dispatch
    await self.on_disconnect(websocket, close_code)
  File "/home/ubuntu/analysis-hub/justpy/justpy.py", line 274, in on_disconnect
    pid = websocket.page_id
AttributeError: 'WebSocket' object has no attribute 'page_id'
```

**Comment by elimintz** [comment]
Thanks for flagging this, I'll take a look. I don't use the whole path for redirection with local, so never encountered this issue.

**Comment by matthewhoendorf** [comment]
I was able to suppress the error with this change to ```justpy/justpy.py```:

```
    async def on_disconnect(self, websocket, close_code):
        try:
            pid = websocket.page_id
        except:
            return
```

**Comment by elimintz** [comment]
Thank you! I made the change to the code.

**Comment by giodegas** [comment]
Has current release fix it?

I still have, sometime in logs:

```python
AttributeError: 'WebSocket' object has no attribute 'page_id'
ERROR websockets_impl: Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 154, in run_asgi
    result = await self.app(self.scope, self.asgi_receive, self.asgi_send)
  File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 146, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/gzip.py", line 20, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 58, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 582, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 299, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/endpoints.py", line 76, in dispatch
    await self.on_disconnect(websocket, close_code)
  File "/usr/local/lib/python3.8/site-packages/justpy/justpy.py", line 274, in on_disconnect
    pid = websocket.page_id
AttributeError: 'WebSocket' object has no attribute 'page_id'

```

**Comment by elimintz** [comment]
Not yet, I haven't issued a new release since. Will do so in the next few weeks.

**Comment by WolfgangFahl** [comment]
has this been fixed or should it be added to 0.3.0?

**Comment by WolfgangFahl** [comment]
Please re-report the error if it is still happening
